### DISCUSSION TITLE
[Bug] Bed respawn point work too strong [Pre8]

### DISCUSSION THREAD

**Discussion by grechnevi** [Pryaxis/TShock#1845] [Q&A]
### **How it should work:**
When you use Magic Mirror it teleports you back to spawn.

### **How it does work:**
1. You click on the bed and "Spawn point set!" text appears in the chat.
2. You use Magic Mirror and you are teleported to the bed.
3. You click on the bed and "**Spawn point removed!**" text appears in the chat.
4. You use Magic Mirror and you are **teleported to the bed**.

### **What I expected to happen:**
1. You click on the bed and "Spawn point removed!" appears
2. Magic Mirror teleports you back to spawn.

**Comment by hakusaro** [comment]
Is this in reference to SSC? This may be fixed in the next pre-release (changed in commit: https://github.com/Pryaxis/TShock/commit/836fd1d156c7b96960bd90a1a3afce47e8c84907).

**Reply by grechnevi** [reply]
Yes, I used SSC while I was testing this.

**Comment by grechnevi** [comment]
Still doesn't work with Pre-release 7.

Also I can't change bed spawn points between two beds instantly, I need to wait 3-5 second.

**Comment by grechnevi** [comment]
Doesn't work with Pre8.

**Comment by hakusaro** [comment]
@AxeelAnder not sure if you have any clues for this?

**Comment by AxeelAnder** [comment]
https://github.com/Pryaxis/TShock/blob/073088156cb28b0d30e1cd6badae9103849ff285/TShockAPI/GetDataHandlers.cs#L2112-L2142

In this case, `force teleport 'vanilla spawn'` and `force teleport phase 1` both will be triggered, and the later overrides the former, leads to the incorrect result.

Maybe we should change this `if` to `else if`
https://github.com/Pryaxis/TShock/blob/073088156cb28b0d30e1cd6badae9103849ff285/TShockAPI/GetDataHandlers.cs#L2121

**Comment by hakusaro** [comment]
Can y'all test this? https://github.com/Pryaxis/TShock/actions/runs/117511286

**Reply by grechnevi** [reply]
Does work right and instantly with setting and removing spawn point.
Does work right and slow (a couple of seconds) with changing spawn points between beds.

**Reply by grechnevi** [reply]
I think that this slowness while changing spawn points can be abused in certain game situations.
You can still teleport to the previous bed and then after some amount of time to the last one.

Also, you can teleport between them **infinitely** by setting spawn point and teleporting instantly, wait some time and do the same.

**Comment by kiroushi** [comment]
Did this break the ability to set the bed as the spawn point with SSC?

DB spawnX and spawnY are not updated when right clicking and setting the spawn for me, no matter which privileges.

**Reply by hakusaro** [reply]
It's entirely possible -- if you can open an issue about this and show the actual in game behavior that'd be nice.
